app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -321,5 +349,32 @@ def run_operation(parameters, operation_name = nil)
         end
       end
     end
+
+    def self.process_old_assignments_ids(old_assignments)
+      old_assignments_ids = []
+      unless old_assignments.empty?
+        if old_assignments[0].class.name == "MiqEnterprise"
```

Comment:
alert_profile has an special use case called "Enterprise" in old_assignments is declared with the "MiqEnterprise" word.
In this case there is not a explicit list of ids, but this word, that means that the real id should be fetched from the database.